mentions-generator/src/main/java/de/brandwatch/minianalytics/mentiongenerator/kafka/Consumer.java

```diff
@@ -1,16 +1,103 @@
 package de.brandwatch.minianalytics.mentiongenerator.kafka;
 
+import de.brandwatch.minianalytics.mentiongenerator.model.Mention;
+import de.brandwatch.minianalytics.mentiongenerator.model.Resource;
+import de.brandwatch.minianalytics.mentiongenerator.postgres.model.Query;
+import de.brandwatch.minianalytics.mentiongenerator.postgres.repository.QueryRepository;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.TextField;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.RAMDirectory;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.kafka.annotation.KafkaListener;
 
+import java.io.IOException;
+import java.util.List;
+
 public class Consumer {
 
     private static final Logger logger = LoggerFactory.getLogger(Consumer.class);
 
+    private QueryRepository queryRepository;
+
+    private Producer producer;
+
+    private Directory memoryIndex = new RAMDirectory();
+    private StandardAnalyzer analyzer = new StandardAnalyzer();
+
+    private IndexWriterConfig indexWriterConfig = new IndexWriterConfig(analyzer);
+    private IndexWriter writer = new IndexWriter(memoryIndex, indexWriterConfig);
+
+    private IndexSearcher searcher;
+    private IndexReader reader;
+
+    private QueryParser queryParser = new QueryParser("text", analyzer);
+
+    @Autowired
+    public Consumer(QueryRepository queryRepository, Producer producer) throws IOException {
+        this.queryRepository = queryRepository;
+        this.producer = producer;
+    }
+
     @KafkaListener(topics = "twitter")
-    public void receive(String message){
-        logger.info("received message='{}'", message);
+    public void receive(Resource resource) throws ParseException, IOException {
+        logger.info("received message='{}'", resource.toString());
```

Comment:
In the long run we discussed to have this method just put the resources on some form of queue and then run matching in batches. Not necessarily for this PR though, but maybe you should create a ticket for that.

Reply:
Can you just confirm once you've seen this and created a ticket?

Reply:
Or just click the resolve button 🤷‍♂

Reply:
Yes, I've seen it. I created a Project Board Card: https://github.com/Max-Leopold/mini-analytics/projects/1#card-26051051